### PR TITLE
Support TOS exception handle in Alluxio

### PR DESCRIPTION
**Pull request by thu-david**
### What changes are proposed in this pull request?

Modified the exception handling related to Tos.

### Why are the changes needed?

1. Because the exception handling needs to conform to Alluxio's conventions.
2. Some exceptions were not handled previously.

### CODE REVIEW

File: underfs/tos/src/main/java/alluxio/underfs/tos/TOSOutputStream.java [hunk 1]
```diff
@@ -167,7 +167,7 @@ public void close() throws IOException {
       return;
     }
     mLocalOutputStream.close();
-    try (BufferedInputStream in = new BufferedInputStream(new FileInputStream(mFile))) {
+    try (BufferedInputStream in = new BufferedInputStream(Files.newInputStream(mFile.toPath()))) {
```
Comment (Jackson-Wang-7) [inline underfs/tos/src/main/java/alluxio/underfs/tos/TOSOutputStream.java:170]: Why did you change the implementation here?

File: underfs/tos/src/main/java/alluxio/underfs/tos/TOSUnderFileSystemFactory.java [hunk 2]
```diff
@@ -40,13 +41,12 @@ public UnderFileSystem create(String path, UnderFileSystemConfiguration conf) {
     if (checkTOSCredentials(conf)) {
       try {
         return TOSUnderFileSystem.createInstance(new AlluxioURI(path), conf);
-      } catch (Exception e) {
+      } catch (TosException e) {
```
Comment (Jackson-Wang-7) [inline underfs/tos/src/main/java/alluxio/underfs/tos/TOSUnderFileSystemFactory.java:44]:
```suggestion
      } catch (TosException e) {
        throw AlluxioTosException.from(e);
      } catch (Exception e) {
```

File: underfs/tos/src/main/java/alluxio/underfs/tos/AlluxioTosException.java [hunk 3]
```diff
@@ -26,27 +26,23 @@ public class AlluxioTosException extends AlluxioRuntimeException {
...
    * @return alluxio tos exception
    */
-  public static AlluxioTosException from(String errorMessage, TosClientException cause) {
-    Status status = Status.UNKNOWN;
-    String errorDescription = "ClientException:" + cause.getMessage();
-    status = httpStatusToGrpcStatus(cause.getStatusCode());
-    errorDescription = cause.getCode() + ":" + cause.getMessage();
+  public static AlluxioTosException from(String errorMessage, TosException cause) {
+    Status status = httpStatusToGrpcStatus(cause.getStatusCode());
+    String errorDescription = cause.getCode() + ":" + cause.getMessage();
```
Comment (Jackson-Wang-7) [inline underfs/tos/src/main/java/alluxio/underfs/tos/AlluxioTosException.java:45]: LOG.warn(errorDescription);

File: underfs/tos/src/main/java/alluxio/underfs/tos/TOSOutputStream.java (hunk 1 above)
Comment (thu-david) [reply underfs/tos/src/main/java/alluxio/underfs/tos/TOSOutputStream.java:170]: Because this way is recommended in effective java.

File: underfs/tos/src/main/java/alluxio/underfs/tos/TOSUnderFileSystemFactory.java (hunk 2 above)
Comment (thu-david) [reply underfs/tos/src/main/java/alluxio/underfs/tos/TOSUnderFileSystemFactory.java:44]: OK, I have revised this function.

**Review by Jackson-Wang-7 — APPROVED** [review]
LGTM

**Comment by Jackson-Wang-7** [comment]
alluxio-bot, merge this please

**Comment by Jackson-Wang-7** [comment]
alluxio-bot, merge this please
